src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const nState = Object.assign({}, state);
+  const nStateArray = [];
+
+  for (const act of actions) {
+    switch (act.type) {
+      case 'clear':
+        Object.keys(nState).forEach(key => {
+          delete nState[key];
+        });
+        break;
+      case 'addProperties':
+        Object.assign(nState, act.extraData);
+        break;
+      case 'removeProperties':
+        act.keysToRemove.forEach(element => {
+          delete nState[element];
+        });
+        break;
+      default:
+        return 'wrong action in actions';
```

Comment:
use `throw new Error('error description');` for error handling

Reply:
ok